src/App.tsx

```diff
@@ -7,23 +7,72 @@ import TaskDashboard from './components/TaskDashboard';
 
 function App() {
   const [taskList, setTaskList] = useState<ITask[]>([]);
+  const [userInput, setUserInput] = useState<string>('');
+  const [isEditing, setIsEditing] = useState<string>('');
 
-  function addNewTask(taskContent: string) {
+  function addTask() {
     const newTask = {
-      task: taskContent,
+      task: userInput,
       completed: false,
       id: String(new Date().valueOf()),
     };
     setTaskList([...taskList, newTask]);
   }
 
+  function removeTask(taskID: string) {
+    setTaskList([...taskList].filter((task) => task.id !== taskID));
+  }
+
+  function startEditing(taskID: string) {
+    setUserInput([...taskList].find((task) => task.id === taskID)?.task ?? '');
+    setIsEditing(taskID);
+  }
+
+  function editTask(input: string, taskID: string) {
+    setTaskList(
+      [...taskList].map((task) => {
+        const taskReturn = task;
+        if (taskReturn.id === taskID) {
+          taskReturn.task = input;
+        }
+        return taskReturn;
+      }),
+    );
+    setIsEditing('');
+    setUserInput('');
+  }
+
+  function checkTaskBox(taskID: string) {
+    setTaskList(
+      [...taskList].map((task) => {
+        const taskReturn = task;
+        if (taskReturn.id === taskID) {
+          taskReturn.completed = !taskReturn.completed;
+        }
+        return taskReturn;
```

Comment:
Não entendi o motivo de ter reassinalado task pra uma nova variavel. Talvez poderiamos usar o spread aqui
```suggestion
        if (task.id === taskID) {
          return { ...task, completed: !task.completed };
        }
        return task
```

Reply:
é pq usando o asing (=)  tava dando erro de objeto not assignable